File: code/switch-scope/sample1.go
package main

import "fmt"

func main() {
	v := 1
	switch v {
	case 1:
		say := "yes"
		fmt.Println("say", say)
	case 2:
		say := "no"
		fmt.Println("say", say)
	default:
		say := "???"
		fmt.Println("say", say)
	}
}

File: code/switch-scope/Sample1.java
public class Sample1 {
	public static void main(String[] args) {
		int v = 1;
		switch (v) {
			case 1: {
				String say = "yes";
				System.out.println("say "+say); 
				}
				break;
			case 2: {
				String say = "no";
				System.out.println("say "+ say); 
				}
				break;
			default: {
				String say = "???";
				System.out.println("say "+ say); 
				}
				break;
		}
	}
}
